cpp/include/raft/linalg/map.cuh

```diff
@@ -22,119 +22,267 @@
 
 #include <raft/core/device_mdspan.hpp>
 #include <raft/core/device_resources.hpp>
-#include <raft/util/input_validation.hpp>
-#include <thrust/tabulate.h>
 
-namespace raft {
-namespace linalg {
+namespace raft::linalg {
 
 /**
- * @brief CUDA version of map
- * @tparam InType data-type upon which the math operation will be performed
- * @tparam MapOp the device-lambda performing the actual operation
- * @tparam TPB threads-per-block in the final kernel launched
- * @tparam Args additional parameters
- * @tparam OutType data-type in which the result will be stored
- * @param out the output of the map operation (assumed to be a device pointer)
- * @param len number of elements in the input array
- * @param map the device-lambda
- * @param stream cuda-stream where to launch this kernel
- * @param in the input array
- * @param args additional input arrays
+ * @defgroup map Mapping ops
+ * @{
+ */
+
+/**
+ * @brief Map a function over zero or more input mdspans of the same size.
+ *
+ * The algorithm applied on `k` inputs can be described in a following pseudo-code:
+ * @code
+ *  for (auto i: [0 ... out.size()]) {
+ *    out[i] = f(in_0[i], in_1[i], ..., in_k[i])
+ *  }
+ * @endcode
+ *
+ * _Performance note_: when possible, this function loads the argument arrays and stores the output
+ * array using vectorized cuda load/store instructions. The size of the vectorization depends on the
+ * size of the largest input/output element type and on the alignment of all pointers.
+ *
+ * Usage example:
+ * @code{.cpp}
+ *  #include <raft/core/device_mdarray.hpp>
```

Comment:
Sorry, I missed this one. This should be `device_mdspan` for the view layer (in all other docs too)

Reply:
But in this example I'm using only functions from `core/mdarray.hpp` (make_device_vector(), .view()), hence I imported that one instead of mdspan.

Reply:
Ah, I missed that- I'm so used to seeing `make_device_vector_view` and `make_device_matrix_view` used in the examples (which I think is fine too). I'm not nitpicky about that and I think it's fine. I'll approve.